src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -0,0 +1,681 @@
+            Cid,
+            Cid,
+            0,
+            nullptr,
+            PacketNumber,
+            BufferLength,
+            Buffer,
+            &PayloadLengthOffset,
+            &PacketNumberLength);
+    if (*PacketLength + CryptoBufferLength > BufferLength) {
+        printf("Crypto Too Big!\n");
+        exit(0);
+    }
+
+    QuicVarIntEncode2Bytes(
+        PacketNumberLength + CryptoBufferLength + CXPLAT_ENCRYPTION_OVERHEAD,
+        Buffer + PayloadLengthOffset);
+    *HeaderLength = *PacketLength;
+
+    CxPlatCopyMemory(Buffer + *PacketLength, CryptoBuffer, CryptoBufferLength);
+    *PacketLength += CryptoBufferLength;
+    *PacketLength += CXPLAT_ENCRYPTION_OVERHEAD;
+}
+
+
+void fuzzInitialPacket(CXPLAT_SOCKET* Binding, CXPLAT_ROUTE Route) {
+    const StrBuffer InitialSalt("afbfec289993d24c9e9786f19c6111e04390a899");
+    const uint16_t DatagramLength = QUIC_MIN_INITIAL_LENGTH; 
+    uint64_t StartTimeMs = CxPlatTimeMs64();
+    int64_t PacketCount = 0;
+    int64_t TotalByteCount = 0;
+    while (CxPlatTimeDiff64(StartTimeMs, CxPlatTimeMs64()) < RunTimeMs) {
+        CXPLAT_SEND_CONFIG SendConfig = { &Route, DatagramLength, CXPLAT_ECN_NON_ECT, 0 };
+        CXPLAT_SEND_DATA* SendData = CxPlatSendDataAlloc(Binding, &SendConfig);
+        if (!SendData){
+            printf("CxPlatSendDataAlloc failed\n");
+        }
+        while (CxPlatTimeDiff64(StartTimeMs, CxPlatTimeMs64()) < RunTimeMs && !CxPlatSendDataIsFull(SendData)) {
+            const uint64_t PacketNumber = 0; //fuzz
+            uint8_t Packet[512] = {0};
+            uint16_t PacketLength, HeaderLength;
+
+            WriteClientInitialPacket(
+                PacketNumber,
+                sizeof(uint64_t),
+                sizeof(Packet),
+                Packet,
+                &PacketLength,
+                &HeaderLength);
+
+            uint8_t *PacketBuffer = Packet;
+            QUIC_LONG_HEADER_V1* Header = (QUIC_LONG_HEADER_V1*)Packet;
+            Header->IsLongHeader = GetRandom(2);
+            Header->FixedBit = GetRandom(2);
+            Header->Reserved = GetRandom(2);
+            Header->PnLength = GetRandom(sizeof(uint32_t));
+            Header->DestCidLength  = (uint8_t)GetRandom(sizeof(uint64_t));
+
+            PacketBuffer += sizeof(QUIC_LONG_HEADER_V1) + sizeof(uint64_t); // point to the source id length
+            *PacketBuffer = (uint8_t)GetRandom(sizeof(uint64_t)); // fuzz the source id length
+
+            uint16_t PacketNumberOffset = HeaderLength - sizeof(uint32_t);
+
+            uint64_t* DestCid = (uint64_t*)(Packet + sizeof(QUIC_LONG_HEADER_V1));
+            uint64_t* SrcCid = (uint64_t*)(Packet + sizeof(QUIC_LONG_HEADER_V1) + sizeof(uint64_t) + sizeof(uint8_t));
+
+            uint64_t* OrigSrcCid = nullptr;
+            for (uint16_t i = HeaderLength; i < PacketLength; ++i) {
+                if (MagicCid == *(uint64_t*)&Packet[i]) {
+                    OrigSrcCid = (uint64_t*)&Packet[i];
+                }
```

Comment:
Apparently Linux doesn't like this unaligned access. Need to use `memcmp`:
```suggestion
                if (!memcmp(&MagicCid, Packet+i, sizeof(MagicCid)) {
                    OrigSrcCid = (uint64_t*)&Packet[i];
                }
```